modules/optimization/include/reporters/OptimizationData.h

```diff
@@ -53,4 +68,8 @@ class OptimizationData : public GeneralReporter
   std::map<std::string, std::vector<Real> *> _weight_names_weights_map;
   /// helper to check data sizes
   void errorCheckDataSize();
+  /// Reference to the FEProblemBase for this helper object
+  FEProblemBase & _fe_problem;
+  /// Thread ID of this helper object
+  const THREAD_ID _tid;
```

Comment:
Do you need these? The parent class should have them, right?

Reply:
Not needed nice catch.